addons/EasyNuke/EasyNuke.lua

```diff
@@ -0,0 +1,207 @@
+spell_tables["thunder"] = T{"Thunder","Thunder II","Thunder III","Thunder IV","Thunder V","Thunder VI",}
+spell_tables["thunder"]["ga"] = T{"Thundaga","Thundaga II","Thundaga III","Thundaja",}
+spell_tables["thunder"]["ra"] = T{"Thundara","Thundara II","Thundara III"}
+spell_tables["light"] = T{"Banish","Banish II","Holy","Banish III",}
+spell_tables["light"]["ga"] = T{"Banishga","Banishga II"}
+spell_tables["dark"] = T{"Impact"}
+spell_tables["dark"]["ga"] = T{"Comet"}
+spell_tables["cure"] = T{"Cure","Cure II","Cure III","Cure IV","Cure V","Cure VI"}
+spell_tables["cure"]["ga"] = T{"Curaga","Curaga II","Curaga III","Curaga IV","Curaga V",}
+spell_tables["cure"]["ra"] = T{"Cura","Cura II","Cura III"} 
+spell_tables["drain"] = T{"Aspir","Aspir II","Aspir III","Drain","Drain II","Drain III"}
+spell_tables["absorb"] = T{"Absorb-Acc","Absorb-TP","Absorb-Attri","Absorb-STR","Absorb-DEX","Absorb-VIT","Absorb-AGI","Absorb-INT","Absorb-MND","Absorb-CHR"}
+
+function find_index ()
+		if current_element == "fire" then elements_index = 1 end
+		if current_element == "wind" then elements_index = 2 end
+		if current_element == "thunder" then elements_index = 3 end
+		if current_element == "light" then elements_index = 4 end
+		if current_element == "ice" then elements_index = 5 end
+		if current_element == "water" then elements_index = 6 end
+		if current_element == "earth" then elements_index = 7 end
+		if current_element == "dark" then elements_index = 8 end
+end
+
+windower.register_event('addon command', function (command, arg)
+
+	if command == 'boom' then
+		if arg == nil then arg = 1 end
+		arg = tonumber(arg)
+		if arg > #spell_tables[current_element] then windower.send_command("@input /echo Invalid Spell. Try again.") return end
+		local current_spell_table = spell_tables[current_element]
+		windower.send_command("@input /ma "..current_spell_table[arg].." <"..target_mode..">")
+	end
+	
+	if command == "boomga" then
+		if arg == nil then arg = 1 end
+		arg = tonumber(arg)
+		if arg > #spell_tables[current_element]["ga"] then windower.send_command("@input /echo Invalid Spell. Try again.") return end
+		local current_spell_table = spell_tables[current_element]["ga"]
+		windower.send_command("@input /ma "..current_spell_table[arg].." <"..target_mode..">")
+	end
+	
+	if command == "boomra" then
+		if arg == nil then arg = 1 end
+		arg = tonumber(arg)
+		if arg > #spell_tables[current_element]["ra"] then windower.send_command("@input /echo Invalid Spell. Try again.") return end
+		local current_spell_table = spell_tables[current_element]["ra"]
+		windower.send_command("@input /ma "..current_spell_table[arg].." <"..target_mode..">")
+	end
+	
+	if command == "target" then
+		if arg then
+			arg = string.lower(arg)
+			target_mode = arg
+			windower.send_command("@input /echo Target mode is now: "..target_mode)
+		else targets_index = targets_index + 1
+		if targets_index > 3 then targets_index = 1 end
```

Comment:
This process, increasing by one and then wrapping the number after a certain limit is very commonly used in this addon and can be simplified:
```lua
targets_index = targets_index % 3 + 1
```
`% 3` tells it to "wrap" around `3`. So if the value is `3` it would wrap to `0` and then add `1`, resulting in `1`. You can use it similarly in other places below, replacing `3` by whatever limit you use there.

Reply:
How would I go about wrapping around a descending order, like I do for the "cycle back" command/argument?

Reply:
```lua
targets_index = (targets_index + 1) % 3 + 1
```

More generally:
```lua
x = (x + N - 2) % N + 1
```
will cycle `x` backwards in the interval `[1, N]`